bridge-token/src/lib.rs

```diff
@@ -44,7 +46,17 @@ pub trait ExtBridgeTokenFactory {
 impl BridgeToken {
     #[init]
     pub fn new() -> Self {
-        assert!(!env::state_exists(), "Already initialized");
+        let current_account_id = env::current_account_id();
+        let (_eth_address, factory_account) = current_account_id
+            .as_str()
+            .split_once(".")
+            .unwrap_or_else(|| env::panic_str("Invalid token address"));
+
+        require!(
+            env::predecessor_account_id().as_str() == factory_account,
+            "Only the factory account can init this contract"
+        );
```

Comment:
Yes, only the factory can create it, but anyone can call new/init because it is a public method.
btw, I have an idea of how this could be fixed in another way, by call the deeply and new in batch transaction